File: split/fatalerror.pas
procedure fatalerror(s: strnumber);
begin
  normalizeselector;
  begin
    if interaction = 3 then;
    printnl(262);
    print(287);
  end;
  begin
    helpptr := 1;
    helpline[0] := s;
  end;
  begin
    if interaction = 3 then
      interaction := 2;
    if logopened then
      error;
    {if interaction>0 then debughelp;}
history := 3;
    jumpout;
  end;
end;

File: cpp/src/fatalerror.cpp
#include "fatalerror.h"
#include "normalizeselector.h"
#include "impression.h"
#include "error.h"
#include "jumpout.h"
#include "texte.h"

void fatalerror(const std::string &s)
{
	normalizeselector();
	print_err("Emergency stop");
	helpptr = 1;
	helpline[0] = s;
	if (interaction == error_stop_mode)
		interaction = scroll_mode;
	if (logopened)
		error();
	history = fatal_error_stop;
	jumpout();
}
